java/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.java

```diff
@@ -233,41 +237,39 @@ private void setupSymbols() {
     //[DocRef: Name=Picture Marker Symbol Drawable-android, Category=Fundamentals, Topic=Symbols and Renderers]
     //Create a picture marker symbol from an app resource
     BitmapDrawable startDrawable = (BitmapDrawable) ContextCompat.getDrawable(this, R.drawable.ic_source);
-    final PictureMarkerSymbol pinSourceSymbol;
     try {
-      pinSourceSymbol = PictureMarkerSymbol.createAsync(startDrawable).get();
-      pinSourceSymbol.loadAsync();
-      pinSourceSymbol.addDoneLoadingListener(new Runnable() {
+      mPinSourceSymbol = PictureMarkerSymbol.createAsync(startDrawable).get();
+      mPinSourceSymbol.loadAsync();
+      mPinSourceSymbol.addDoneLoadingListener(new Runnable() {
         @Override
         public void run() {
           //add a new graphic as start point
           mSourcePoint = new Point(-117.15083257944445, 32.741123367963446, SpatialReferences.getWgs84());
-          Graphic pinSourceGraphic = new Graphic(mSourcePoint, pinSourceSymbol);
+          Graphic pinSourceGraphic = new Graphic(mSourcePoint, mPinSourceSymbol);
           mGraphicsOverlay.getGraphics().add(pinSourceGraphic);
         }
       });
-      pinSourceSymbol.setOffsetY(20);
+      mPinSourceSymbol.setOffsetY(20);
```

Comment:
I think this is another case where loadable chaining would work.

something like

```
pinSourceSymbol = PictureMarkerSymbol.createAsync(startDrawable).get();
      pinSourceSymbol.setOffsetY(20);
      //add a new graphic as start point
      mSourcePoint = new Point(-117.15083257944445, 32.741123367963446, SpatialReferences.getWgs84());
      Graphic pinSourceGraphic = new Graphic(mSourcePoint, pinSourceSymbol);
      mGraphicsOverlay.getGraphics().add(pinSourceGraphic);
    } catch (InterruptedException | ExecutionException e) {
      String error = "Error creating picture marker symbol: " + e.getMessage();
      Log.e(TAG, error);
      Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
```

Reply:
this is a really nice refactor for readability as well!